raster_analysis/data_environment.py

```diff
@@ -0,0 +1,174 @@
+# flake8: noqa
+import json
+from collections import Iterable, defaultdict
+from copy import deepcopy
+from typing import Any, Callable, Dict, List, Optional, Type, Union, cast
+
+from numpy import (
+    ceil,
+    datetime64,
+    float,
+    float32,
+    float64,
+    floor,
+    timedelta64,
+    uint,
+    uint16,
+    uint32,
+    uint64,
+)
+from pandas import Series
+from pydantic import BaseModel, parse_obj_as
+
+from raster_analysis.globals import BasePolygon
+from raster_analysis.grid import Grid, GridName, TileScheme
+
+
+class SourceLayer(BaseModel):
+    source_uri: str
+    name: str
+    tile_scheme: TileScheme = TileScheme.nw
+    grid: GridName = GridName.ten_by_forty_thousand
+    pixel_encoding: Dict[Any, Any] = {}
+    decode_expression: str = ""
+    encode_expression: str = ""
+
+
+class DerivedLayer(BaseModel):
+    source_layer: str
+    name: str
+    derivation_expression: str
```

Comment:
just a detail, but in pixetl we call a similar parameter `calc`. Not saying that this is a better name but we should try to stay consistent across repos

Reply:
I think that name should be fine here, it's already called DerivedLayer so kind of redundant anyway